### PR TITLE
Support IPv6 in routing procedure responses

### PR DESCRIPTION
**Pull request by lutovich**
This PR improves parsing in `BoltServerAddress` to support IPv6 in `getServers()` and `getRoutingTable()` procedure responses.

### CODE REVIEW

File: driver/src/test/java/org/neo4j/driver/internal/net/BoltServerAddressTest.java [hunk 1]
```diff
@@ -50,4 +51,37 @@ public void shouldAlwaysResolveAddress()
...
+        assertEquals( "127.0.0.1", address1.host() );
+        assertEquals( 1234, address1.port() );
+
+        BoltServerAddress address2 = new BoltServerAddress( "8.8.8.8:8080" );
+        assertEquals( "8.8.8.8", address2.host() );
+        assertEquals( 8080, address2.port() );
+    }
+
+    @Test
+    public void shouldParseIPv6Addresses()
```
Comment (martinfurmanski) [inline driver/src/test/java/org/neo4j/driver/internal/net/BoltServerAddressTest.java]: Could you try also with zone-id like [fe80::b279:2f%en0]:1234 ?

File: driver/src/test/java/org/neo4j/driver/internal/net/SocketConnectionTest.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public void shouldReceiveServerInfoAfterInit() throws Throwable
         SocketClient socket = mock( SocketClient.class );
         SocketConnection conn = new SocketConnection( socket, SERVER_INFO, DEV_NULL_LOGGER );
 
-        when( socket.address() ).thenReturn( BoltServerAddress.from( URI.create( "http://neo4j.com:9000" ) ) );
+        when( socket.address() ).thenReturn( new BoltServerAddress( URI.create( "http://neo4j.com:9000" ) ) );
```
Comment (martinfurmanski) [inline driver/src/test/java/org/neo4j/driver/internal/net/SocketConnectionTest.java]: Unrelated, but why is this http:// ?

**Review by martinfurmanski — APPROVED** [review]

**Comment by martinfurmanski** [comment]
In the server code I handle the hostname string representation as only ::1 and not as [::1]. Only when combined in some form are the brackets used, e.g. bolt://[::1]:1234 or in config [fe80::1]:1234 etc. Just for consideration.

**Comment by lutovich** [comment]
@martinfurmanski comments should now be addressed

**Review by ali-ince — APPROVED** [review]

**Comment by lutovich** [comment]
@martinfurmanski, @ali-ince thanks for the review! Merging...

**Pull request merged**